src/gwt/src/org/rstudio/studio/client/workbench/views/source/model/DocUpdateSentinel.java

```diff
@@ -327,12 +328,16 @@ private boolean doSave(String path,
          }
       }
       
-      // Update marks after document save
+      final boolean saved = didSave;
```

Comment:
My main concern with doing this on save is that autosaves happen half a second after the edit, which is plenty of time to move your cursor. So the position we're going to save and restore is neither (a) where you made your last edit, nor (b) where you last left your cursor, but "somewhere near" your last edit, which may not feel predictable.